pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -382,14 +382,27 @@ def _resolve_sampler(self, dataloader: DataLoader, shuffle: bool, mode: Optional
                     " distributed training. Either remove the sampler from your DataLoader or set"
                     " `replace_sampler_ddp=False` if you want to use your custom sampler."
                 )
-            return self._get_distributed_sampler(
+            sampler = self._get_distributed_sampler(
                 dataloader,
                 shuffle,
                 mode=mode,
                 overfit_batches=self.trainer.overfit_batches,
                 **self.trainer.distributed_sampler_kwargs,
             )
 
+            # update docs too once this is resolved
+            trainer_fn = self.trainer.state.fn
+            if isinstance(sampler, DistributedSampler) and trainer_fn in (TrainerFn.VALIDATING, TrainerFn.TESTING):
+                rank_zero_warn(
```

Comment:
I would make info. This is definitely a critical behavior for the users. This would be removed when we support uneven ddp.

Reply:
I would do only test for now and prevent DistributedSampler injection for testing.

Reply:
I'd still consider using warning since this is critical, and an info might not reflect how important it is for benchmarking.

why not for `trainer.validate`? it uses the exact same API as that of `test`

Reply:
+1 for warning